packages/api-docs-builder/ApiBuilders/HookApiBuilder.ts

```diff
@@ -264,12 +264,17 @@ const attachTable = (
       const requiredProp = prop.required;
 
       const deprecation = (propDescriptor.description || '').match(/@deprecated(\s+(?<info>.*))?/);
-
+      const typeDescription = (propDescriptor.typeStr ?? '')
+        .replace(/&/g, '&amp;')
+        .replace(/</g, '&lt;')
+        .replace(/>/g, '&gt')
+        .replace(/"/g, '&quot;')
+        .replace(/'/g, '&#39;');
       return {
         [propName]: {
           type: {
-            name: propDescriptor.typeStr,
-            description: propDescriptor.description ?? undefined,
+            name: typeDescription,
```

Comment:
In the components, the `type.name` refers to: `shape`, `union` etc, while the description contains the processed type. This is specific to the structure that react-docgen generates. In order to match the same and simplify things, I've added the type string to both the name and the description for the hooks' types.

Reply:
Can you leave this as a comment in code? Here maybe, https://github.com/mui/material-ui/pull/36168/files#r1104474099

Reply:
Good point, added!